kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ParquetConverters.java

```diff
@@ -145,10 +141,9 @@ public abstract static class BasePrimitiveColumnConverter
         }
 
         @Override
-        public boolean moveToNextRow() {
-            resizeIfNeeded();
+        public void moveToNextRow(long prevRowIndex) {
             currentRowIndex++;
-            return this.nullability[currentRowIndex - 1];
+            resizeIfNeeded();
```

Comment:
Don't we call resizeIfNeeded() before each addValue? I feel like we should do this either here or there but not both

Reply:
Also renamed to the `moveToNextRow()` to `finalizeCurrentRow` which is what exactly it is doing.

Reply:
Do we have the caller calling getColumnarBatch/getColumnVector before hitting initialBatchSize as an invariant? If so maybe we should at least document that. Because before I think we had it so that it would still work